core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/store/dot-favorite-page.store.ts

```diff
@@ -0,0 +1,159 @@
+import { Injectable } from '@angular/core';
+import { DotRolesService } from '@dotcms/app/api/services/dot-roles/dot-roles.service';
+import { DotRole } from '@dotcms/app/shared/models/dot-role/dot-role.model';
+import { ComponentStore, tapResponse } from '@ngrx/component-store';
+import { forkJoin, Observable, of, throwError } from 'rxjs';
+import { switchMap, take } from 'rxjs/operators';
+import { DotFavoritePageFormData, DotFavoritePageProps } from '../dot-favorite-page.component';
+import { DotMessageService } from '@dotcms/app/api/services/dot-message/dot-messages.service';
+import { HttpErrorResponse } from '@angular/common/http';
+import { DotHttpErrorManagerService } from '@dotcms/app/api/services/dot-http-error-manager/dot-http-error-manager.service';
+import { DotTempFileUploadService } from '@dotcms/app/api/services/dot-temp-file-upload/dot-temp-file-upload.service';
+import { DotWorkflowActionsFireService } from '@dotcms/app/api/services/dot-workflow-actions-fire/dot-workflow-actions-fire.service';
+import { DotCMSContentlet, DotCMSTempFile } from '@dotcms/dotcms-models';
+import { DotCurrentUserService } from '@dotcms/app/api/services/dot-current-user/dot-current-user.service';
+import { DotCurrentUser } from '@dotcms/app/shared/models/dot-current-user/dot-current-user';
+
+export interface DotFavoritePageState {
+    roleOptions: DotRole[];
+    currentUserRoleId: string;
+    pageRenderedHtml?: string;
+    isAdmin: boolean;
+    imgWidth: number;
+    imgHeight: number;
+    loading: boolean;
+    closeDialog: boolean;
+}
+
+export const enum LoadingState {
+    INIT = 'INIT',
+    LOADING = 'LOADING',
+    LOADED = 'LOADED'
+}
```

Comment:
We should unify `LoadingState` and loading being a boolean, makes no sense to have two, check this post: https://indepth.dev/posts/1033/ngrx-how-and-where-to-handle-loading-and-error-states-of-ajax-calls